SDL/Event.cu

```diff
@@ -1128,13 +1126,14 @@ void SDL::Event::createSegmentsWithModuleMap()
     if(segmentsInGPU == nullptr)
     {
         segmentsInGPU = (SDL::segments*)cms::cuda::allocate_host(sizeof(SDL::segments), stream);
-        createSegmentsInExplicitMemory(*segmentsInGPU, N_MAX_SEGMENTS_PER_MODULE, nLowerModules, N_MAX_PIXEL_SEGMENTS_PER_MODULE,stream);
+        unsigned int nTotalSegments;
+        createSegmentArrayRanges(*modulesInGPU, *rangesInGPU, *mdsInGPU, nLowerModules, nTotalSegments, stream, N_MAX_PIXEL_SEGMENTS_PER_MODULE);
```

Comment:
This line was not here in the previous version of the code. Why did it have to appear now?

Reply:
Because we want to get the nTotalSegments from this function. Then we can put this into createSegmentsInExplicitMemory

Reply:
But now I see that we also run this function in another place in the code (L765 of this file). Does it really make sense to "`createSegmentArrayRanges`" twice? Isn't there a way to get the information from the first running of the function?